newIDE/app/src/EventsSheet/ParameterFields/AnyVariableField.js

```diff
@@ -0,0 +1,124 @@
+import {
+  type ParameterFieldProps,
+  type ParameterFieldInterface,
+  type FieldFocusFunction,
+} from './ParameterFieldCommons';
+import { enumerateVariablesOfContainersList } from './EnumerateVariables';
+import { mapFor } from '../../Utils/MapFor';
+
+export default React.forwardRef<ParameterFieldProps, ParameterFieldInterface>(
+  function SceneVariableField(props: ParameterFieldProps, ref) {
+    const field = React.useRef<?VariableFieldInterface>(null);
+    const [editorOpen, setEditorOpen] = React.useState(false);
+    const focus: FieldFocusFunction = options => {
+      if (field.current) field.current.focus(options);
+    };
+    React.useImperativeHandle(ref, () => ({
+      focus,
+    }));
+
+    const {
+      project,
+      scope,
+      instruction,
+      onInstructionTypeChanged,
+      projectScopedContainers,
+    } = props;
+    const { layout } = scope;
+
+    const enumerateGlobalAndSceneVariables = React.useCallback(
+      () =>
+        enumerateVariablesOfContainersList(
+          projectScopedContainers.get().getVariablesContainersList()
+        ),
+      [projectScopedContainers]
+    );
+
+    const variablesContainers = React.useMemo(
+      () => {
+        const variablesContainersList = projectScopedContainers
+          .get()
+          .getVariablesContainersList();
+        return mapFor(
+          0,
+          variablesContainersList.getVariablesContainersCount(),
+          i => {
+            return variablesContainersList.getVariablesContainer(i);
+          }
+        );
+      },
+      [projectScopedContainers]
+    );
+
+    const isGlobal = !!(
+      layout &&
+      project &&
+      !layout.getVariables().has(getRootVariableName(props.value)) &&
+      project.getVariables().has(getRootVariableName(props.value))
+    );
+
+    return (
+      <React.Fragment>
+        <VariableField
+          forceDeclaration
+          project={project}
+          instruction={instruction}
+          variablesContainers={variablesContainers}
+          enumerateVariables={enumerateGlobalAndSceneVariables}
+          parameterMetadata={props.parameterMetadata}
+          value={props.value}
+          onChange={props.onChange}
+          isInline={props.isInline}
+          onRequestClose={props.onRequestClose}
+          onApply={props.onApply}
+          ref={field}
+          onOpenDialog={() => setEditorOpen(true)}
+          globalObjectsContainer={props.globalObjectsContainer}
+          objectsContainer={props.objectsContainer}
+          projectScopedContainers={projectScopedContainers}
+          scope={scope}
+          id={
+            props.parameterIndex !== undefined
+              ? `parameter-${props.parameterIndex}-scene-variable-field`
+              : undefined
+          }
+          onInstructionTypeChanged={onInstructionTypeChanged}
+        />
+        {editorOpen && layout && project && (
+          <GlobalAndSceneVariablesDialog
```

Comment:
In theory this could be a "GlobalAndSceneAndMultipleLocalVariablesDialog" right?
But for now, we say we don't care?

(Important)

Reply:
No, I think we won't make a dialog with 3 tabs.
A list of all local variables (from the edited event) would be out of context without the event sheet and it would be hard for users to make sense of them.
I think users should only be able to edit local variables by clicking on them from the event sheet.

Reply:
Ok! 👍